nifi-framework-bundle/nifi-framework/nifi-framework-components/src/main/java/org/apache/nifi/util/FlowDifferenceFilters.java

```diff
@@ -112,6 +133,15 @@ private static ComponentNode getComponent(final FlowManager flowManager, final C
 
     }
 
+    private static ComponentNode getComponent(final FlowManager flowManager, final VersionedComponent component) {
+        if (!(component instanceof InstantiatedVersionedComponent)) {
+            return null;
+        }
+
+        final InstantiatedVersionedComponent instantiatedComponent = (InstantiatedVersionedComponent) component;
+        return getComponent(flowManager, component.getComponentType(), instantiatedComponent.getInstanceIdentifier());
```

Comment:
Minor implementation note, it looks like this could be adjusted to use `instanceof InstantiatedVersionedComponent instantiated) { ...` and otherwise return `null`, instead of the short-circuit return and casting.